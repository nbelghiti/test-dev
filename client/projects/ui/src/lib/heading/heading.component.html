<div
	class="c-heading {{ classes }}"
	[ngClass]="{'c-heading--inline': inline}"
>
	<ng-container [ngSwitch]="level">
		<h1 *ngSwitchCase="HeadingLevel.heading1">{{title}}</h1>
		<h2 *ngSwitchCase="HeadingLevel.heading2">{{title}}</h2>
		<h3 *ngSwitchCase="HeadingLevel.heading3">{{title}}</h3>
	</ng-container>

	<div class="c-heading__meta">
		<i
			*ngIf="loading; else withContent"
			class="fas fa-{{IconName.spinner}} fa-spin c-heading__loader"
		>
		</i>

		<ng-template #withContent>
			<ng-content></ng-content>
		</ng-template>
	</div>
</div>
