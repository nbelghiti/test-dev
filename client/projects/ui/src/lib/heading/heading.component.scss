// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Heading
// =============================================================================
.c-heading {
	display: inline-flex;
	flex-direction: column;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-heading__loader {
	color: $primary-color;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-heading--inline {
	flex-direction: row;
	align-items: baseline;

	.c-heading__meta {
		margin-top: 0;
		margin-left: 1rem;
	}
}
