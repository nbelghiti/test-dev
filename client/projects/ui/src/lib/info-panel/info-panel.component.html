<div class="c-info-panel">
	<header class="c-info-panel__header">
		<ui-heading
			class="c-info-panel__title"
			level=3
			[title]="title"
		>
		</ui-heading>
	</header>

	<div
		#content
		*ngIf="hasContent"
		class="c-info-panel__content"
	>
		<ng-content></ng-content>
	</div>
</div>
