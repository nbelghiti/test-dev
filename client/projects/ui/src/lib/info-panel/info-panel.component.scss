// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Info Panel
// =============================================================================
.c-info-panel {
	width: 100%;
	padding: 2rem 0;

	background-color: $background-color-base;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-info-panel__header {
	padding: 0 2rem;
}

.c-info-panel__content {
	margin-top: 1rem;
}
