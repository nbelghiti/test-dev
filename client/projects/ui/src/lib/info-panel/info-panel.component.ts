import { AfterViewInit, Component, ElementRef, Input, ViewChild } from '@angular/core';


@Component({
	selector: 'ui-info-panel',
	templateUrl: './info-panel.component.html',
	styleUrls: ['./info-panel.component.scss']
})

export class InfoPanelComponent implements AfterViewInit {
	@Input() public title: string;

	@ViewChild('content', {static: false}) contentChild: ElementRef<any>;
	public hasContent: boolean = true;

	public ngAfterViewInit(): void {
		setTimeout(() => {
			if (!this.contentChild) { return; }

			this.hasContent = this.contentChild.nativeElement.childNodes.length ? true : false;
		});
	}
}
