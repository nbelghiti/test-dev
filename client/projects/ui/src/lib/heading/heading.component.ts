import { Component, Input } from '@angular/core';

import { IconName } from '../icon/icon.types';

import { HeadingLevel } from './heading.types';

@Component({
	selector: 'ui-heading',
	templateUrl: './heading.component.html',
	styleUrls: ['./heading.component.scss']
})

export class HeadingComponent {
	@Input() public title: string;
	@Input() public classes: string = '';
	@Input() public inline: boolean = false;
	@Input() public loading: boolean = false;
	@Input() public level: HeadingLevel = HeadingLevel.heading1;

	public HeadingLevel: typeof HeadingLevel = HeadingLevel;
	public IconName: typeof IconName = IconName;
}
