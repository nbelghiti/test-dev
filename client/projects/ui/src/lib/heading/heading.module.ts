import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { HeadingComponent } from './heading.component';

@NgModule({
	imports: [
		CommonModule,
	],
	declarations: [
		HeadingComponent,
	],
	exports: [
		HeadingComponent,
	],
})
export class HeadingModule {}
