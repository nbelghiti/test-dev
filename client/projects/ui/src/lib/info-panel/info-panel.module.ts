import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { HeadingModule } from '../heading/heading.module';

import { InfoPanelComponent } from './info-panel.component';

@NgModule({
	imports: [
		CommonModule,
		HeadingModule,
	],
	declarations: [
		InfoPanelComponent,
	],
	exports: [
		InfoPanelComponent,
	],
})
export class InfoPanelModule {}
