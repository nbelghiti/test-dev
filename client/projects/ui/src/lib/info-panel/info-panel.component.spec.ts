import { ComponentFixture, TestBed, async } from '@angular/core/testing';

import { HeadingComponent } from '../heading/heading.component';

import { InfoPanelComponent } from './info-panel.component';

describe('InfoPanelComponent', () => {
	let component: InfoPanelComponent;
	let fixture: ComponentFixture<InfoPanelComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			declarations: [ InfoPanelComponent, HeadingComponent ]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(InfoPanelComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
