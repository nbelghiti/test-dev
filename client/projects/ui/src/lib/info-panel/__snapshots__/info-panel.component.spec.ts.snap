// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoPanelComponent Should render correctly 1`] = `
<ui-info-panel
  contentChild={[Function ElementRef]}
  hasContent={[Function Boolean]}
>
  <div
    class="c-info-panel"
  >
    <header
      class="c-info-panel__header"
    >
      <ui-heading
        class="c-info-panel__title"
        level="3"
        ng-reflect-level="3"
      >
        <div
          class="c-heading"
          ng-reflect-klass="c-heading "
          ng-reflect-ng-class="[object Object]"
        >
          
          
          
          
          <h3>
            
          </h3>
          <div
            class="c-heading__meta"
          >
            
            
            
          </div>
        </div>
      </ui-heading>
    </header>
    
    <div
      class="c-info-panel__content"
    />
  </div>
</ui-info-panel>
`;
